packages/block-library/src/navigation-link/edit.js

```diff
@@ -296,6 +361,26 @@ export default function NavigationLinkEdit( {
 		[ clientId ]
 	);
 
+	// Store the colors from context as attributes for rendering
+	useEffect( () => {
+		setAttributes( {
```

Comment:
I think this may be causing undo issues, e.g: if I apply a background overlay color pressing undo does not undo the color as expected. I think the set attributes call is probably creating undo levels.
__unstableMarkNextChangeAsNotPersistent may be helpful in this case.
What makes us need to set the context as attributes? Couldn't we also rely on accessing the context on the server for rendering?

Reply:
Thanks, I think this was vestigial and I was able to simplify it in a414a98989 without having to use `__unstableMarkNextChangeAsNotPersistent`